src/dispatcher/index.ts

```diff
@@ -163,14 +166,15 @@ const createDispatcher = () => {
       prevPayload: string | MetaPayload,
       payload: any
     ) => {
-      if (type === 'title') {
-        titleQueue[titleQueue.indexOf(prevPayload as string)] = payload;
-        if (!isServerSide)
-          document.title = applyTitleTemplate(payload, titleTemplateQueue[0]);
-      } else if (type === 'titleTemplate') {
-        titleTemplateQueue[
-          titleTemplateQueue.indexOf(prevPayload as string)
-        ] = payload;
+      if (type === TITLE || type === TEMPLATE) {
```

Comment:
CC @gustavpursche added some cases here to update head when we update title or template on index 0